01_oop_and_ood/src/Point.ts

```diff
@@ -0,0 +1,36 @@
+export class Point {
+  public x: number;
+  public y: number;
+
+  constructor();
+  constructor(x: number, y: number);
+  constructor(x: number = 0, y: number = 0) {
+    this.x = x;
+    this.y = y;
+  }
+
+  public toString() {
+    return `(${this.x}, ${this.y})`;
+  }
+
+  public distance(): number;
+  public distance(point: Point): number;
+  public distance(x: number, y: number): number;
+  public distance(x?: any, y?: any): number {
```

Comment:
why not `x?: number | Point, y?: number`?

Reply:
changed